pkg/controller/deployment/deployment_controller.go

```diff
@@ -576,6 +579,73 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 	return dc.syncDeploymentStatus(allRSs, newRS, deployment)
 }
 
+// proportionalScaling should run only on scaling events and not during the normal rollout process.
+func (dc *DeploymentController) proportionalScaling(deployment extensions.Deployment, shouldCreate bool) error {
+	newRS, oldRSs, err := dc.getAllReplicaSets(deployment, shouldCreate)
+	if err != nil {
+		return err
+	}
+	allRSs := controller.FilterActiveReplicaSets(oldRSs)
+	maybeMaxSurge := 0
```

Comment:
I think based on scope and usage this is the best place I can have it right now.